package apollo;

message ShipClass {
    required string className = 1;
    required int64 mass = 2;
    optional string folderName = 3;
    optional string meshName = 4;
    optional string description = 5;
}

message ShipClasses {
    repeated ShipClass shipClass = 1;
}

// Optional must be used to allow defaults
message ShipStations {
    optional bool mainScreen = 1 [default = false];
    optional bool navigation = 2 [default = false];
    optional bool weapons = 3 [default = false];
    optional bool engineering = 4 [default = false];
    optional bool science = 5 [default = false];
    optional bool comms = 6 [default = false];
}

message Ship {
    required int32 id = 1;
    optional double x = 2;
    optional double y = 3;
    optional double z = 4;
    optional float h = 5;
    optional float p = 6;
    optional float r = 7;
    optional double dx = 8;
    optional double dy = 9;
    optional double dz = 10;
    optional float dh = 11;
    optional float dp = 12;
    optional float dr = 13;
    optional float thrust = 14;
    optional float torque = 15;
    repeated Turret turrets = 16;
    optional ShipStations stations = 33;
    optional string name = 34;
    optional string className = 35;
}

message Ships {
    repeated Ship ship = 1;
}

message Throttle{
    optional float normal = 1;
    optional float warp = 2;
    optional bool jump = 3;
    optional double bsgjumpx = 4;
    optional double bsgjumpz = 5;
    optional float heading = 6;
}

message Target{
    required int32 targetId = 1;
    required string turretName = 2;
}

message Turret{
    required string turretName = 1;
    repeated Joint joints = 2;
}

message Joint{
    required string jointName = 1;
    required float targetH = 2;
    required float targetP = 3;
    required float targetR = 4;
    required float time = 5;
}